File: Frontend/src/types/reservation.ts
export type ReservationStatus =
  | 'beklemede'
  | 'onaylandı'
  | 'giriş_yapıldı'
  | 'çıkış_yapıldı'
  | 'iptal_edildi';

export interface Reservation {
  id: number;
  room_id: number;
  guest_name: string;
  guest_email: string;
  guest_phone: string;
  guest_id_number: string;
  guest_count: number;
  special_requests: string;
  check_in_date: string;   // ISO date string "2025-06-01"
  check_out_date: string;
  actual_check_in?: string;
  actual_check_out?: string;
  price_per_night: number;
  total_nights: number;
  total_price: number;
  status: ReservationStatus;
  notes: string;
  created_at: string;
  updated_at?: string;
  capacity?: string;
  features?: string;
}

export interface ReservationCreate {
  room_id: number;
  guest_name: string;
  guest_email: string;
  guest_phone: string;
  guest_id_number: string;
  guest_count: number;
  special_requests: string;
  check_in_date: string;
  check_out_date: string;
  notes: string;
}

export interface ReservationUpdate {
  guest_name?: string;
  guest_email?: string;
  guest_phone?: string;
  guest_id_number?: string;
  guest_count?: number;
  special_requests?: string;
  check_in_date?: string;
  check_out_date?: string;
  status?: ReservationStatus;
  notes?: string;
}

export interface AvailabilityCheck {
  room_id: number;
  check_in_date: string;
  check_out_date: string;
}

export interface AvailabilityResponse {
  available: boolean;
  room_id: number;
  check_in_date: string;
  check_out_date: string;
  price_per_night: number;
  total_nights: number;
  total_price: number;
  message: string;
}

export interface ReservationStats {
  total: number;
  beklemede: number;
  'onaylandı': number;
  'giriş_yapıldı': number;
  'çıkış_yapıldı': number;
  'iptal_edildi': number;
}

File: Backend/app/schemas/reservation.py
from pydantic import BaseModel, EmailStr
from typing import Optional
from datetime import date, datetime


class ReservationBase(BaseModel):
    room_id: int
    guest_name: str
    guest_email: str
    guest_phone: str = ""
    guest_id_number: str = ""
    guest_count: int = 1
    special_requests: str = ""
    check_in_date: date
    check_out_date: date
    notes: str = ""


class ReservationCreate(ReservationBase):
    pass


class ReservationUpdate(BaseModel):
    guest_name: Optional[str] = None
    guest_email: Optional[str] = None
    guest_phone: Optional[str] = None
    guest_id_number: Optional[str] = None
    guest_count: Optional[int] = None
    special_requests: Optional[str] = None
    check_in_date: Optional[date] = None
    check_out_date: Optional[date] = None
    status: Optional[str] = None
    notes: Optional[str] = None


class ReservationStatusUpdate(BaseModel):
    status: str  # beklemede, onaylandı, giriş_yapıldı, çıkış_yapıldı, iptal_edildi


class Reservation(ReservationBase):
    id: int
    price_per_night: float
    total_nights: int
    total_price: float
    status: str
    actual_check_in: Optional[datetime] = None
    actual_check_out: Optional[datetime] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class ReservationWithRoom(Reservation):
    room_number: Optional[str] = None
    room_type: Optional[str] = None
    capacity: Optional[str] = None
    features: Optional[str] = None

    class Config:
        from_attributes = True


class AvailabilityCheck(BaseModel):
    room_id: int
    check_in_date: date
    check_out_date: date


class AvailabilityResponse(BaseModel):
    available: bool
    room_id: int
    check_in_date: date
    check_out_date: date
    price_per_night: Optional[float] = None
    total_nights: Optional[int] = None
    total_price: Optional[float] = None
    message: str = ""